File: code/backend/python/app/core/models/auth/guest_info.py
from pydantic import BaseModel, Field, ConfigDict
from typing import Optional

class GuestInfo(BaseModel):
    user_uuid: str = Field(..., alias="userUuid")
    username: str
    nickname: Optional[str] = None
    role: str
    state: int
    guest_cookie_token: str = Field(..., alias="guestCookieToken")
    guest_expires_at: Optional[str] = Field(None, alias="guestExpiresAt")
    language: Optional[str] = None
    ts_registration: Optional[str] = Field(None, alias="tsRegistration")
    ts_last_access: Optional[str] = Field(None, alias="tsLastAccess")
    expired: bool = False

    model_config = ConfigDict(populate_by_name=True)

File: code/backend/java/core/src/main/java/games/paths/core/model/auth/GuestInfo.java
package games.paths.core.model.auth;

/**
 * GuestInfo - Domain model for guest user details visible to administrators.
 * Contains the identity, session status, and timestamps of a guest user.
 * Immutable record.
 */
public record GuestInfo(
        String userUuid,
        String username,
        String nickname,
        String role,
        int state,
        String guestCookieToken,
        String guestExpiresAt,
        String language,
        String tsRegistration,
        String tsLastAccess,
        boolean expired) {

    public GuestInfo {
        if (userUuid == null || userUuid.isBlank()) {
            throw new IllegalArgumentException("userUuid is required");
        }
        if (username == null || username.isBlank()) {
            throw new IllegalArgumentException("username is required");
        }
    }
}
